docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,188 @@
+
+So, any given child tenant can be thought of as the *root of its own tenancy hierarchy*.
+
+Ultimately, there may well be a real “root” tenant somewhere in the tree, but a good tenancy model means we should not have to care whether we are the ultimate root tenant, or some child of a child of a child.
+
+[Tenant roots](Tenant roots.png)
+
+For example, as an Azure customer, we don’t need to think about the fact that Azure has other tenants when we use it to provision and manage our own resources. Nor do we need to theorise about the way in which services like Cosmos DB or SQL Serverless are themselves provisioned and managed as tenants in Azure when we provision and consume resources from them.
+
+Should we wish to provide a multi-tenancy model for our solution, we are free to do so in any way we choose. We are not bound to the tenancy model of our parent.
+
+A common challenge when designing a multi-tenanted system is thinking about the right “scope”. Am I thinking about the solution in my tenant (which should not expose details of tenancy) or the multi-tenancy solution itself (which should not depend on implementation details of the solutions that happen to be deployed as its tenants)? You must take care that one does not bleed into the other.
+
+## Solution plane and management plane
+There is one respect in which the child tenant is aware of the existence of a parent. It needs to be able to *manage its tenancy*. To do so, it will provision, configure, and monitor services through tools provided by the parent that hosts it.
+
+We often refer to the divide between the services you *consume in your tenant* and the services you use to *manage your tenant* as the *solution plane* and the *management plane*.
+
+[Solution v. Management Plane](Solution v Management Plane)
+
+Careful conceptual separation of solution plane and management plane is essential.
+
+## Common requirements of a multi-tenanted solution
+Let’s think about a very generalised application. We might say that a typical application has a stimulus/response design like this:
+
+1. Receive a stimulus of some kind (e.g. HTTP request, event, message, timer)
+1. Connect to and authenticate against one or more services using service-specific credentials, in order to obtain the information required to respond to the stimulus.
+1. Process that information (and optionally respond), possibly connecting to other services to update state related to that request.
+
+There are numerous services you can use with that basic pattern – HTTP requests, queues and message busses, event stores, various flavours of databases and storage, caching. The details of these services is outside the scope of this discussion, but they are all consumed using this basic pattern.
+
+In a multi-tenanted solution, you need to layer a notion of “the intended tenant” into your design:
```

Comment:
One thing that happens in Azure is that these components are not addressable from within the tenant, as this would compromise isolation.

Reply:
I'm not sure I understand this point? Can you explain a bit more?

Reply:
Specifically, I was thinking about all the "software defined networking" components that sit in front of many of the Azure services, which control ingress into the service. For example the scale controller which monitors each request to figure out how to scale up, or in Dapr / ACA that's fulfilled by (managed) KEDA etc...